backend/tracim_backend/config.py

```diff
@@ -131,10 +139,79 @@ def get_raw_config(
         )
         return config_value
 
-    # INFO - G.M - 2019-04-05 - Config loading methods
+    # INFO - G.M - 2019-04-05 - load of enabled app
+    def load_enabled_apps(self) -> None:
+        self._load_enabled_apps_config()
+        loaded_apps = load_apps()
+        self._load_enabled_app(self.APP__ENABLED, loaded_apps)
 
-    def load_config(self) -> None:
-        """Parse configuration file and env variables"""
+    def _load_enabled_apps_config(self) -> None:
+        self.log_config_header("App Enabled config parameters:")
+        default_enabled_app = (
+            "contents/thread,"
+            "contents/file,"
+            "contents/html-document,"
+            "contents/folder,"
+            "{extend_apps}"
+            "share_content,"
+            "upload_permission,"
+            "gallery"
+        )
+        extend_apps = ""
+        # TODO - G.M - 2020-01-09 - remove this retrocompat code, as
+        #  CALDAV__ENABLED and COLLABORATIVE_DOCUMENT_EDITION__ACTIVATED
+        # should be deprecated.
+        self.CALDAV__ENABLED = asbool(self.get_raw_config("caldav.enabled", "false"))
+        if self.CALDAV__ENABLED:
+            extend_apps += ",agenda"
+        self.COLLABORATIVE_DOCUMENT_EDITION__ACTIVATED = asbool(
+            self.get_raw_config("collaborative_document_edition.activated", "false")
+        )
+        if self.COLLABORATIVE_DOCUMENT_EDITION__ACTIVATED:
+            extend_apps += ",collaborative_document_edition"
+        default_enabled_app = default_enabled_app.format(extend_apps=extend_apps)
+        self.APP__ENABLED = string_to_list(
+            self.get_raw_config("app.enabled", default_enabled_app),
+            separator=",",
+            cast_func=str,
+            do_strip=True,
+        )
+
+    def _load_enabled_app(
+        self, enabled_app_slug_list: typing.List[str], loaded_apps: typing.List[TracimApplication]
+    ) -> None:
+
+        # TODO - G.M - 2018-08-08 - [GlobalVar] Refactor Global var
+        # of tracim_backend, Be careful app_list is a global_var
+        app_list.clear()
+        # INFO - G.M - 2020-01-09 - order of app in app_list should be:
+        # - all enabled app according to enabled_app_slug_list order
```

Comment:
This code is complex to preserve order from config file. We should probably deal with order differently. Order is directly associated to entry, so we should probably do this in https://github.com/tracim/tracim/issues/2326.